shared/nomad-scripts/nomad-startup.sh.tpl

```diff
@@ -3,24 +3,13 @@
 # One of 'AWS' or 'GCP'. Value passed into template
 export CLOUD_PROVIDER=${cloud_provider}
```

Comment:
Hmm… You are removing all the conditionals that deal with GCP/AWS differences, but leave this variable in place and keep the file in the `shared` folder. That strikes me as inconsistent - Either keep the conditionals or move the file to the aws module folder and remove this variable.

Reply:
We'll need to add GCP support shortly. Didn't mean to the support here, this is just a copy and paste of the current state in `circleci/server`. I'd like to keep in `/shared` but do you mind if I add back the GCP specific bits in an iteration?

Reply:
Generally no, but experience has taught me that these "I'll add this back in later" kind of things gets lost easily :/

Reply:
Resolved out of band: going to iterate right after this PR with those changes just to unblock the release process.